### ISSUE TITLE
Nested values can't be set individually

### ISSUE THREAD

**Issue by glarizza** [glarizza/puppet-property_list_key#4]
Basically, consider a value property like this:

<pre>
value => {
           [
             {
                'one_key' => 'one_value'
             }
             {
                'two_key' => 'two_value'
             }
             {
                'three_key' => 'three_value'
             }
           [
         }            
</pre>

If you wanted to manage just ONE of those nested dictionaries, you would have to manage the ENTIRE array of dictionaries or else the provider would clobber the entire file.  We need to decide whether we want to just manage individual pieces of a value (probably, yes) or THE ENTIRE VALUE (most likely no).  If you managed the entire value, then, in essence, you would have a key/value for the entire file...and plists are long mmmkay.

**Comment by cdenneen** [comment]
I'm having this same problem with persistent-others or persistent-apps. Having the ability to say add a shortcut to my dock would be nice. But this means I'd need to maintain the entire key or if I use something like array-add it would just continuously add the shortcuts on each run.

**Comment by glarizza** [comment]
As an aside - If I recall correctly, I think I remember seeing @jhaals with a solution in Boxen for setting dock items in a specific order (but, of course, I can't find it now)

**Comment by jhaals** [comment]
I actually gave up trying to do this and started manage the whole dock plist with boxen.
It would be awesome to actually do it properly

**Comment by grahamgilbert** [comment]
I made [this](https://github.com/grahamgilbert/puppet-dockutil) - it uses [dockutil](https://github.com/kcrawford/dockutil), so obviously not useful more generally.

**Comment by jhaals** [comment]
Sweet. Just one thing.
`action   => "add",` could this be `ensure => present` instead? Would be nice to keep it the puppet way.

**Comment by grahamgilbert** [comment]
Yes, have moved this into it's [own issue](https://github.com/grahamgilbert/puppet-dockutil/issues/1)

**Comment by glarizza** [comment]
Thanks @grahamgilbert - that's the thing I was thinking of!

**Comment by cdenneen** [comment]
Thanks guys… yeah actually grabbed dockutil last night.
Graham is there support for the After or Before part vs the Position?

https://github.com/kcrawford/dockutil

**Comment by grahamgilbert** [comment]
Not yet, but file an issue on the repo and I'll see what I can do.

On 4 Apr 2013, at 10:57 PM, cdenneen notifications@github.com wrote:

Thanks guys… yeah actually grabbed dockutil last night.
Graham is there support for the After or Before part vs the Position?

https://github.com/kcrawford/dockutil

On Apr 4, 2013, at 2:05 PM, Gary Larizza notifications@github.com wrote:

> Thanks @grahamgilbert - that's the thing I was thinking of!
> 
> —
> Reply to this email directly or view it on GitHub.

—
Reply to this email directly or view it on
GitHubhttps://github.com/glarizza/puppet-property_list_key/issues/4#issuecomment-15926633
.

**Comment by ahrenstein** [comment]
I'd also like to add my vote for being able to manage nested values individually.

**Comment by jantman** [comment]
Yeah, this is pretty important for me as well.
